deepinv/training/trainer.py

```diff
@@ -253,6 +257,17 @@ def setup_train(self):
 
         self.loss_history = []
 
+    def reorder_losses(self):
```

Comment:
Why is this needed? seems a bit hacky

Reply:
I agree, but I don't see another solution at the moment. Both SplittingLoss and R2R need to save the mask/noise they use in the forward eval `x_net = model(y, physics)` to use it in the loss computation. If another loss is called before them, eg EILoss, it will generate a new mask/noise during the model's call, and this might cause supoptimal training.

Reply:
It makes sense, but I think it's risky (and not very OOP) to have the mask/noise rely on no forward pass being called in between the forward pass and the loss calculation. Could we put `self.mask = get_mask(...` in the `SplittingModel.__init__` so the mask definitely stays the same for forward pass and the loss calculation? For the `tsize` we could ask for it in the `SplittingLoss` definition and pass it to the `SplittingModel`

Reply:
maybe we can use a fix that is robust and doesn't require changing that much?

```
x_net=model(y, physics)
if hasattr(model, update_parameters)
       x_net=model(y, physics, update_params=True)
else: 
      x_net = model(y, physics)
```

Reply:
Looks good to me!